### ISSUE TITLE
variable into a query?

### ISSUE THREAD

**Issue by Fulup** [chriszarate/sheetrock#75]
I must be missing something obvious, but I have not found a way to use a form response to include a variable into the sql query.

<form id="form" name="form">
 <fieldset><legend>Search for someone commemorated in this project</legend> 
  <label for="searchterm">Name:</label> 
  <input id="searchterm" name="searchterm" type="text" value="" /> 
  <input name="submit" type="submit" value="Search" /> 
  <input name="reset" type="reset" value="Reset" />
 </fieldset>
</form>

<table class="table table-condensed table-striped" id="statistics">
</table>

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-sheetrock/1.0.0/dist/sheetrock.min.js"></script>

<script type="text/javascript">

var searchterm = document.getElementById('searchterm');

// document.write(searchterm.value);

var q = searchterm.value;

var mySpreadsheet = 'https://docs.google.com/spreadsheets/d/1_bxcejxQHGRgVnyMEaSYo4fgqXJBTL6gqI2Fwxa9r8M/edit#gid=0';
// Load an entire worksheet, limited by the searchterm.
$('#statistics').sheetrock({
  url: mySpreadsheet,
query: "select A,C,D,E,F where K = '+q+' order by C asc"
 // query: "select A,C,D,E,F where J = 'Individuals' and K = "'+q+'" order by C asc", fetchSize: 10, headersOff: true,
});
</script>

Any help much appreciated.

**Comment by szyam** [comment]
Hey @Fulup wondering if you ever solved this?

**Comment by chriszarate** [comment]
The query (as passed to the `query` option) is just a string, so you will need to retrieve the form value with JavaScript into a variable and construct the string, e.g.:

```js
var name = $('#field').val();
var myQuery = "select A,B,C where A = '" + name + "';"

$("#my-table").sheetrock({
  url: "...",
  query: myQuery
});
```

**Comment by Fulup** [comment]
Hi Chris,
Thanks for your response.

I did use the same approach eventually:

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-sheetrock/1.0.0/dist/sheetrock.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/handlebars.js/1.1.2/handlebars.min.js"></script><script>
$(document).ready(function(){
$("#submit").click(function() {

  var surname = $("#surname").val();
  $.trim(surname);
  var mySpreadsheet = 'https://docs.google.com/spreadsheets/d/1_bxcejxQHGRgVnyMEaSYo4fgqXJBTL6gqI2Fwxa9r8M/edit#gid=0';
// Compile the Handlebars template for HR leaders.
  var HRTemplate = Handlebars.compile($('#hr-template').html());
// Load search spreadsheet.
// A Date, C Headline, D Text, E Media, F Media credit, G Media caption,
// I Type, J Tag, K Surname, L Forenames, M Rank, N Military ID, O CWGC, P Regiment, Q Memorial, R Decoration

  $('#hr').sheetrock({
   url: mySpreadsheet,
   query: "select C,E,F,G,M,N,O,P,R where J = 'Individuals' and K = '"+ $.trim(surname) +"' order by K asc",
   fetchSize: 15,
   rowTemplate: HRTemplate
  });
// Set style of even list items
$("li:even").css('background-color', '#FFF8DC');
});
});
</script>
<h1>Type in a surname</h1>

<p><input id="surname" value="" /><button id="submit">Search</button></p>

<p id="output">&nbsp;</p>

<ul id="hr"><script id="hr-template" type="text/x-handlebars-template">
    <li>
      <h3>{{cellsArray.[0]}}
      {{#if cellsArray.[8]}}
        , {{cellsArray.[8]}}
      {{/if}}
      </h3>
      {{cellsArray.[4]}}, {{cellsArray.[7]}}, {{cellsArray.[5]}}<br />
      {{#if cellsArray.[1]}}
        <a href="{{cellsArray.[1]}}">
        <img src="{{cellsArray.[1]}}" width="550" height="412" alt="Piece embroidered in memory of {{cellsArray.[0]}}" />
        </a>
      {{/if}}
      <p class="italic">{{cellsArray.[2]}}</p>
      <p>{{cellsArray.[3]}}</p>
      {{#if cellsArray.[6]}}
        <p><a href="http://www.cwgc.org/find-war-dead/casualty/{{cellsArray.[6]}}/">More about {{cellsArray.[0]}} from the
Commonwealth War Graves Commission website</a>.</p>
      {{/if}}
    </li>
  </script>
</ul>

Philip Adams
Senior Assistant Librarian: Content Delivery, Department of Library and Learning Services

DE MONTFORT UNIVERSITY
Leicester LE1 9BH
T: +44 (0) 116 250 6397
E:  pka@dmu.ac.uk<mailto:pka@dmu.ac.uk>
W: dmu.ac.uk
UNISON Equality Coordinator at  DMU
Join Unison: http://bit.ly/t1nUsq
